conda_smithy/tests/test_feedstock_io.py

```diff
@@ -0,0 +1,710 @@
+        filename = os.path.join(self.tmp_dir, filename)
+
+        write_text = "text"
+        with fio.write_file(filename) as fh:
+            fh.write(write_text)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual(write_text, read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual(write_text, read_text)
+
+
+    def test_write_file_2(self):
+        filename = "test.txt"
+
+        write_text = "text"
+        with fio.write_file(filename) as fh:
+            fh.write(write_text)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual(write_text, read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual(write_text, read_text)
+
+
+    def test_write_file_3(self):
+        filename = "dir1/dir2/test.txt"
+
+        write_text = "text"
+        with fio.write_file(filename) as fh:
+            fh.write(write_text)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual(write_text, read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual(write_text, read_text)
+
+
+    def test_touch_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        fio.touch_file(filename)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual("", read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual("", read_text)
+
+
+    def test_touch_file_2(self):
+        filename = "test.txt"
+
+        fio.touch_file(filename)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual("", read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual("", read_text)
+
+
+    def test_touch_file_3(self):
+        filename = "dir1/dir2/test.txt"
+
+        fio.touch_file(filename)
+
+        read_text = ""
+        with open(filename, "r") as fh:
+            read_text = fh.read()
+
+        self.assertEqual("", read_text)
+
+        blob = next(self.repo.index.iter_blobs(BlobFilter(filename)))[1]
+        read_text = blob.data_stream[3].read().decode("utf-8")
+
+        self.assertEqual("", read_text)
+
+
+    def test_remove_file_1(self):
```

Comment:
`test_remove_file_1` and `test_remove_file_2` are basically the same since `os.chdir(self.tmp_dir)` is done during `setUp`, so can we remove one of them?